### PR TITLE
fix(cli): panic with `deno coverage`

### PR DESCRIPTION
**Pull request by 0xIchigo**
This PR directly addresses the issue raised in #23282 where Deno panics if `deno coverage` is called with `--include` regex that returns no matches.

I've opted not to change the return value of `collect_summary` for simplicity and return an empty `HashMap` instead

### CODE REVIEW

**Comment by bartlomieju** [comment]
Thanks for the PR, could you please add a test case that exercises this change? You can add an `itest!` check in `tests/integration/coverage_tests.rs`

**Comment by 0xIchigo** [comment]
My apologies @bartlomieju — I missed your initial reply to this PR. I've gone ahead and added a test at `tests/integration/coverage_tests.rs`

File: tests/integration/coverage_tests.rs [hunk 1]
```diff
@@ -555,3 +555,64 @@ File         | Branch % | Line % |
...
+  output.assert_exit_code(1);
+
+  let actual: &str = output.combined_output();
+  let expected_message: &str = "error: No test modules found";
+  assert!(
+    actual.contains(expected_message),
+    "Expected output to contain '{}', but found:\n{}",
+    expected_message,
+    actual
+  );
```
Comment (bartlomieju) [inline tests/integration/coverage_tests.rs:588]: You can use `assert_contains!` macro here to skip manually creating a message

File: tests/integration/coverage_tests.rs [hunk 2]
```diff
@@ -555,3 +555,64 @@ File         | Branch % | Line % |
...
+      eprintln!("{:?}", path);
+    }
+  }
+
+  // Assert that the coverage directory is otherwise empty
+  assert!(
+        unexpected_contents.is_empty(),
+        "Expected the coverage directory to be empty except for 'empty_dir', but found: {:?}",
+        unexpected_contents
+    );
```
Comment (bartlomieju) [inline tests/integration/coverage_tests.rs:615]: It's strange that `rustfmt` didn't dedent this... oh well

File: tests/integration/coverage_tests.rs [hunk 3]
```diff
@@ -555,3 +555,64 @@ File         | Branch % | Line % |
...
+
+  println!("No unexpected coverage files found as expected, test concludes successfully.");
```
Comment (bartlomieju) [inline tests/integration/coverage_tests.rs:617]: Maybe skip this line, it won't get printed by default anyway

**Review by bartlomieju — COMMENTED** [review]
Looks good, thanks for the PR. Could you please use `assert_contains!` macro?

**Review by bartlomieju — APPROVED** [review]
LGTM, thanks!

**Pull request merged**